examples/templates-minijinja/src/main.rs

```diff
@@ -0,0 +1,86 @@
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p example-templates-minijinja
+//! ```
+//! Demo for the MiniJinja templating engine.
+//! Exposes three pages all sharing the same layout with a minimal nav menu.
+
+use axum::http::StatusCode;
+use axum::{response::Html, routing::get, Extension, Router};
+use minijinja::{context, Environment};
+use std::sync::Arc;
+
+#[tokio::main]
+async fn main() {
+    // init template engine and add templates
+    let mut env = Environment::new();
+    env.add_template("layout", include_str!("../templates/layout.jinja"))
+        .unwrap();
+    env.add_template("home", include_str!("../templates/home.jinja"))
+        .unwrap();
+    env.add_template("content", include_str!("../templates/content.jinja"))
+        .unwrap();
+    env.add_template("about", include_str!("../templates/about.jinja"))
+        .unwrap();
+    let env = Arc::new(env);
+
+    // define routes
+    let app = Router::new()
+        .route("/", get(handler_home))
+        .route("/content", get(handler_content))
+        .route("/about", get(handler_about))
+        .layer(Extension(env)); // add state as extension
```

Comment:
I think it would be nicer to use `with_state` and the `State` extractor instead of `Extension` for this.